libtenzir/include/tenzir/detail/string.hpp

```diff
@@ -67,105 +69,127 @@ trim(std::string_view value, const std::string_view whitespace
   return value;
 }
 
-/// @brief Checks whether the index `idx` in `text` is escaped
-inline auto is_escaped(size_t idx, std::string_view text) {
-  if (idx >= text.size()) {
-    return false;
-  }
-  // An odd number of preceding backslashes means it is escaped, for example:
-  // `x\n` => true, `x\\n` => false, `x\\\n` => true, 'x\\\\n' => false.
-  auto backslashes = size_t{0};
-  while (idx > 0 and text[idx - 1] == '\\') {
-    ++backslashes;
-    --idx;
-  }
-  return backslashes % 2 == 1;
-}
-
-/// finds the index of the first occurrence that is not enclosed my matching
-/// quotes quotes that are not closed are not considered quoting anything
-/// @param s the string to search
-/// @param find a list of characters to search for
-/// @param quotes list of characters to consider as "quotes"
-/// @param start index to start the search at
-/// @returns index of the first occurrence of a character from `find` that
-/// is`not enclosed by matching `quotes`; `npos` otherwise
-inline auto
-find_first_of_not_in_quotes(std::string_view s, std::string_view find,
-                            size_t start = 0,
-                            std::string_view quotes = "\"\'") -> size_t {
-  size_t quote_start = s.npos;
-  size_t find_pos = s.npos;
-  const auto is_quote_at = [&](size_t i) {
-    if (quotes.find(s[i]) == quotes.npos) {
-      return false;
-    }
-    return not is_escaped(i, s);
-  };
-
-  for (size_t i = start; i < s.size(); ++i) {
-    if (is_quote_at(i)) {
-      if (quote_start == s.npos) {
-        quote_start = i;
-        continue;
-      } else if (s[quote_start] == s[i]) {
-        quote_start = s.npos;
-        continue;
-      }
-    }
-    if (find.find(s[i]) != s.npos) {
-      if (quote_start == s.npos) {
-        return i;
-      }
-      bool is_enclosed = false;
-      for (size_t j = i + 1; j < s.size(); ++j) {
-        if (is_quote_at(j) and s[j] == s[quote_start]) {
-          is_enclosed = true;
-          break;
-        }
-      }
-      if (not is_enclosed) {
-        return i;
-      }
+inline auto is_escaped_at(std::string_view text, size_t i) -> bool {
+  auto slashes = size_t{0};
```

Comment:
nit: A slash is not a backslash ;)

Also, since we removed the check from the previous version, does it make sense to assert here that `i < text.size()`?